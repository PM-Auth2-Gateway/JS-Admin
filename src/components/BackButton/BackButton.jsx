import { useHistory } from 'react-router-dom';
import { Button } from 'react-bootstrap';

const BackButton = () => {
  const history = useHistory();
  return (
    <Button variant='secondary' onClick={history.goBack}>
      Back
    </Button>
  );
};

export default BackButton;
