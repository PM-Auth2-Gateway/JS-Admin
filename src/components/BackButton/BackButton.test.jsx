import { shallow } from 'enzyme';
import { useHistory } from 'react-router-dom';
import { Button } from 'react-bootstrap';

import BackButton from './BackButton';

jest.mock('react-router-dom', () => ({
  useHistory: jest.fn(),
}));

describe('BackButton Component', () => {
  const historyGoBack = jest.fn();

  beforeEach(() => {
    useHistory.mockReturnValue({
      goBack: historyGoBack,
    });
  });

  it('should render', () => {
    const component = shallow(<BackButton />);

    expect(component).toMatchSnapshot();
  });

  it('should go back on click', () => {
    const component = shallow(<BackButton />);

    const btn = component.find(Button);
    btn.simulate('click');

    expect(historyGoBack).toBeCalledTimes(1);
  });
});
