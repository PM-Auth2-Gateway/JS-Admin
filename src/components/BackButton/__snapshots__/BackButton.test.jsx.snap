// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton Component should render 1`] = `
<Button
  active={false}
  disabled={false}
  onClick={[MockFunction]}
  variant="secondary"
>
  Back
</Button>
`;
